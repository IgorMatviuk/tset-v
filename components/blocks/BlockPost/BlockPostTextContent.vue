<template>
  <div class="post-text-content">
    <BContainer>
      <div class="post-text-content__wrapper" v-html="content"></div>
    </BContainer>
  </div>
</template>

<script>
export default {
  props: {
    content: {
      type: String,
      required: true
    }
  }
}
</script>

<style lang="scss">
.post-text-content {
  margin-bottom: 25px;

  &__wrapper {
    p {
      margin-bottom: 0;
      font-size: 12px;
      line-height: 15px;
      color: $input-color;

      &:not(:last-child) {
        margin-bottom: 10px;
      }
    }

    h2,
    h3,
    h4 {
      margin-bottom: 15px;
      font-size: 13px;
      font-weight: 500;
      line-height: 16px;
      color: $black;
      text-align: center;
    }
  }

  @include media-breakpoint-up(md) {
    &__wrapper {
      width: 450px;
      margin-right: auto;
      margin-left: auto;

      p {
        font-size: 13px;
        line-height: 16px;
      }

      h2,
      h3,
      h4 {
        margin-bottom: 12px;
        font-size: 14px;
        line-height: 18px;
      }
    }
  }

  @include media-breakpoint-up(lg) {
    &__wrapper {
      width: 610px;

      p {
        font-size: 14px;
        line-height: 18px;
      }

      h2,
      h3,
      h4 {
        margin-bottom: 15px;
      }
    }
  }

  @include media-breakpoint-up(xl) {
    margin-bottom: 30px;

    &__wrapper {
      width: 770px;
    }
  }

  @include media-breakpoint-up(xxl) {
    &__wrapper {
      width: 930px;
    }
  }
}
</style>
