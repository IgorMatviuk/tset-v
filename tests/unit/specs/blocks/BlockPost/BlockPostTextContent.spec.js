import { mount, createLocalVue } from '@vue/test-utils'
import _cloneDeep from 'lodash.clonedeep'

import { BContainer } from 'bootstrap-vue'

import BlockPostTextContent from '@/components/blocks/BlockPost/BlockPostTextContent'

import createFactory from '@/api/helpers/factory'
import postAttrs from '@/api/factories/post'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)

describe('BlockPostTextContent', () => {
  let textContent

  function createBlockPostTextContent() {
    const wrapper = mount(BlockPostTextContent, {
      localVue,
      propsData: {
        content: _cloneDeep(textContent)
      }
    })

    return wrapper
  }

  beforeEach(() => {
    const post = createFactory(postAttrs, 1)
    const textBlock = post[0].blocks.find((item) => item.type === 'text')

    textContent = textBlock.content
  })

  test('render posts text', () => {
    const wrapper = createBlockPostTextContent()

    expect(wrapper.element).toMatchSnapshot()
  })
})
