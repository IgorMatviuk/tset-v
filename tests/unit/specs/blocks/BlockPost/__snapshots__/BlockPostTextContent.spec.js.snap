// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockPostTextContent render posts text 1`] = `
<div
  class="post-text-content"
>
  <div
    class="container"
  >
    <div
      class="post-text-content__wrapper"
    >
      <p>
        Последнее время, мы всё чаще слышим фразу «Made in Ukraine», она приобрела совсем новый смысл. Внезапно, стереотип “наше – значит плохо” сменился на полную противоположность. И дело не только в патриотизме! А в чем? 
        <b>
          Почему стоит выбирать именно украинский бренд?
        </b>
         Почему мы всё реже заглядываем в магазины известных масс-маркетов? Читай дальше!
      </p>
    </div>
  </div>
</div>
`;
